src/main/java/org/apache/commons/imaging/formats/png/PngWriter.java

```diff
@@ -513,25 +497,36 @@ public void writeImage(final BufferedImage src, final OutputStream os, Map<Strin
                         (byte) 1);
             }
         }
-
-        if (params.containsKey(ImagingConstants.PARAM_KEY_XMP_XML)) {
-            final String xmpXml = (String) params.get(ImagingConstants.PARAM_KEY_XMP_XML);
+        
+        // care about XMP XML if provided by parameter 
+        if (parameters.isXmpXmlAsStringPresent()) {
+            final String xmpXml = parameters.getXmpXmlAsString();
             writeChunkXmpiTXt(os, xmpXml);
         }
-
-        if (params.containsKey(PngConstants.PARAM_KEY_PNG_TEXT_CHUNKS)) {
-            final List<?> outputTexts = (List<?>) params.get(PngConstants.PARAM_KEY_PNG_TEXT_CHUNKS);
-            for (Object outputText : outputTexts) {
-                final PngText text = (PngText) outputText;
-                if (text instanceof PngText.Text) {
-                    writeChunktEXt(os, (PngText.Text) text);
-                } else if (text instanceof PngText.Ztxt) {
-                    writeChunkzTXt(os, (PngText.Ztxt) text);
-                } else if (text instanceof PngText.Itxt) {
-                    writeChunkiTXt(os, (PngText.Itxt) text);
-                } else {
-                    throw new ImageWriteException(
-                            "Unknown text to embed in PNG: " + text);
+        
+        // text chunks are parameters specific for PNG images
+        // so we need to ask first if parameters specific for PNG images are provided
+        // than ask for text chunks
+        if (parameters instanceof ImagingParametersPng) {
```

Comment:
Would be good if generics could be used here to avoid the `instanceof`, or another way to model objects.